src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -1178,6 +1178,10 @@ void format(RevCommit commit, @Nullable RevCommit parent, PrintWriter pw, Boolea
         }
     }
 
+    public CleanCommand clean_() {
+    	throw new UnsupportedOperationException("No JGit support!");
```

Comment:
The current implementation of JGit does not support exclusion, or I am overlooking something... The only option I see is an inclusion list for paths (exact match, no regex). Probably it is possible to calculate the diff and feed JGit the entire list of paths it is allowed to clean, but that sounds like a real ugly workaround for a missing option in JGit imo.